yarn-project/simulator/src/client/view_data_oracle.ts

```diff
@@ -321,4 +321,12 @@ export class ViewDataOracle extends TypedOracle {
 
     await this.db.removeNullifiedNotes(this.contractAddress);
   }
+
+  public override store(key: Fr, values: Fr[]): Promise<void> {
+    return this.db.store(this.contractAddress, key, values);
+  }
+
+  public override load(key: Fr): Promise<Fr[]> {
+    return this.db.load(this.contractAddress, key);
```

Comment:
Let's make `contractAddress` a param and assert that it matches `this.contractAddress`, with a TODO linking to https://github.com/AztecProtocol/aztec-packages/issues/10727 (same as I did in #10651) and add a comment in that issue mentioning this. If a separate contract needs to process notes, they'll also need to be able to access the db.

Reply:
https://github.com/AztecProtocol/aztec-packages/pull/10867/commits/b63f6a3d885ffc4b67f90159b3263a612fe98e35